src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,225 @@
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+    """
+    with expectation:
+        assert orthology_benchmark.find_latest_core(core_names) == exp_output
+
+
+@pytest.mark.parametrize(
+    "multi_dbs",
+    [
+        [{'src': 'core/danio_rerio_core_105_11', 'name': 'danio_rerio_core_105_11'},
+         {'src': 'core/mus_musculus_cbaj_core_107_1', 'name': 'mus_musculus_cbaj_core_107_1'},
+         {'src': 'core/mus_musculus_core_106_39', 'name': 'mus_musculus_core_106_39'}]
+    ],
+    indirect=True
+)
+class TestGetCoreNames:
+    """Tests :func:`orthology_benchmark.get_core_names()` function.
+
+    Attributes:
+        core_dbs: A set of test core databases.
+
+    """
+
+    core_dbs = {} # type: Dict
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, multi_dbs: Dict) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            multi_dbs: Dictionary of unit test databases (fixture).
+
+        """
+        type(self).core_dbs = multi_dbs
+
+    @pytest.mark.parametrize(
+        "species_names, exp_output, expectation",
+        [
+            (["danio_rerio", "mus_musculus", "zea_mays"],
+             {"danio_rerio": os.environ['USER'] + "_" + "danio_rerio_core_105_11",
+             "mus_musculus": os.environ['USER'] + "_" + "mus_musculus_core_106_39", "zea_mays": ""},
+             does_not_raise()),
+            ([], None, raises(RuntimeError,
+                              match=r"Empty list of species names. Cannot search for core databases."))
+        ]
+    )
+    def test_get_core_names(self, species_names: List[str], exp_output: Dict[str, str],
+                            expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.get_core_names()` function
+        when connection to the server host can be established.
+
+        Args:
+            species_names: Species (genome) names.
+            exp_output: Expected return value of the function.
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                    exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            host = self.core_dbs["mus_musculus_core_106_39"].dbc.host
+            port = self.core_dbs["gallus_gallus_core_99_6"].dbc.port
+            # user "travis" hardcoded until we find a better solution
+            assert orthology_benchmark.get_core_names(species_names, host, port, "travis") == exp_output
+
+    def test_get_core_names_fake_connection(self) -> None:
+        """Tests :func:`orthology_benchmark.get_core_names()` function
+        when provided fake server connection details."""
```

Comment:
Another docstring summary line suggestion:
```suggestion
        """Tests :func:`orthology_benchmark.get_core_names()` with fake server details."""
```